orbit/cmd/desktop/desktop.go

```diff
@@ -21,6 +25,55 @@ import (
 // version is set at compile time via -ldflags
 var version = "unknown"
 
+func setupRunners() {
+	var runnerGroup run.Group
+
+	// Setting up a watcher for the communication channel
+	if runtime.GOOS == "windows" {
+		runnerGroup.Add(
+			func() error {
+				// block wait on the communication channel
+				if err := blockWaitForStopEvent(constant.DesktopAppExecName); err != nil {
+					log.Error().Err(err).Msg("There was an error on the desktop communication channel")
+					return err
+				}
+
+				log.Info().Msg("Shutdown was requested!")
+				return nil
+			},
+			func(err error) {
+				systray.Quit()
+			},
+		)
+	}
+
+	// Setting up a signal handler runner
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var execFn func() error
+	var interrFn func(error)
+
+	if runtime.GOOS == "windows" {
+		execFn, interrFn = run.SignalHandler(ctx, os.Interrupt, os.Kill)
+	} else {
+		execFn, interrFn = run.SignalHandler(ctx, os.Interrupt, os.Kill, syscall.SIGTERM)
+	}
+
+	runnerGroup.Add(
+		execFn,
+		func(err error) {
+			interrFn(err)
+			systray.Quit()
```

Comment:
Curious why this is needed (if we already call it in the other interruptFn in line 45), is this for `darwin` and `linux`?

Reply:
yes, this will ensure that systray.Quit() gets called when the application is signaled. In the case of windows it works when the fleet-desktop is compiled as a console application (without the ldflags "-H=windowsgui"). The signal handler logic should also work on linux and darwin although I have not tried it.